### PR TITLE
Fixed issue #19163: Reflected XSS in HtmlEditorPop - Bypassing XSS sanitization function

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/admin/HtmlEditorPop.php [hunk 1]
```diff
@@ -35,7 +35,11 @@ public function index()
             $aData['sControlIdDis'] = $aData['sFieldName'] . '_popupctrldis';
             $aData['toolbarname'] = 'popup';
             $aData['htmlformatoption'] = '';
-            $aData['contentsLangDirection'] = sanitize_xss_string(App()->request->getQuery('contdir'));
+            $contentsLangDirection = App()->request->getQuery('contdir');
+            if (!in_array(strtolower((string) $contentsLangDirection), ['ltr', 'rtl'])) {
+                $contentsLangDirection = null;
```
Comment (Shnoulle) [inline application/controllers/admin/HtmlEditorPop.php:40]: ? Then we don't have `contentsLangDirection = ;` in js ?

We don't have lang in query ?
If not : set to ltr

**Review by Shnoulle — COMMENTED** [review]
Best is to set a value in my option, it must be a string.

File: application/controllers/admin/HtmlEditorPop.php (hunk 1 above)
Comment (gabrieljenik) [reply application/controllers/admin/HtmlEditorPop.php]: Solved

**Review by Shnoulle — APPROVED** [review]

**Comment by Shnoulle** [comment]
Front end seems broken, not related to this commit.

Need testing

File: application/controllers/admin/HtmlEditorPop.php [hunk 2]
```diff
@@ -35,7 +35,11 @@ public function index()
             $aData['sControlIdDis'] = $aData['sFieldName'] . '_popupctrldis';
             $aData['toolbarname'] = 'popup';
             $aData['htmlformatoption'] = '';
-            $aData['contentsLangDirection'] = sanitize_xss_string(App()->request->getQuery('contdir'));
+            $contentsLangDirection = App()->request->getQuery('contdir');
+            if (!in_array(strtolower((string) $contentsLangDirection), ['ltr', 'rtl'])) {
+                $contentsLangDirection = getLanguageRTL(Yii::app()->session['adminlang']) ? 'rtl' : 'ltr';
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/controllers/admin/HtmlEditorPop.php:40]: Sorry, use "what", "where" and "how"? :)

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/admin/HtmlEditorPop.php:40]: What : using `$contentsLangDirection = getLanguageRTL(Yii::app()->session['adminlang']) ? 'rtl' : 'ltr';` always (no need param)

But made an error : direction are not dependent from admin language , but from text language.

[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/controllers/admin/HtmlEditorPop.php:40]: I see.
I will leave it as it is as to not create unexpected problems :)

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/admin/HtmlEditorPop.php:40]: Oh yes, i really don't want new update here, can be merged !

**Pull request merged**